### PR TITLE
Move functions into DebugReportData

### PR DESCRIPTION
**Pull request by spencer-lunarg**
On the quest to improve logging, first need to do some house cleaning

This change

1. `typedef struct debug_report_data` -> `class DebugReportData`
2. Move things into `DebugReportData` instead of passing around a pointer reference to it

I imagine a lot of this stems back from when there were multiple layers, but instead of functions like `BeginCmdDebugUtilsLabel` being globals that pass in a `debug_report_data*`, I just moved these calls into the new `DebugReportData` class

### CODE REVIEW

File: layers/error_message/logging.h [hunk 1]
```diff
@@ -178,76 +178,30 @@ class TypedHandleWrapper {
     VulkanTypedHandle handle_;
 };
 
-typedef struct _debug_report_data {
+struct Location;
+
+class DebugReportData {
+  public:
     std::vector<VkLayerDbgFunctionState> debug_callback_list;
-    VkDebugUtilsMessageSeverityFlagsEXT active_severities{0};
```
Comment (spencer-lunarg) [inline layers/error_message/logging.h:183]: these I found are private as no one else touches them

File: layers/error_message/logging.h [hunk 2]
```diff
@@ -260,67 +214,56 @@ typedef struct _debug_report_data {
...
+    void EndCmdDebugUtilsLabel(VkCommandBuffer command_buffer);
+    void InsertCmdDebugUtilsLabel(VkCommandBuffer command_buffer, const VkDebugUtilsLabelEXT *label_info);
+    void ResetCmdDebugUtilsLabel(VkCommandBuffer command_buffer);
+    void EraseCmdDebugUtilsLabel(VkCommandBuffer command_buffer);
 
-struct Location;
-VKAPI_ATTR bool LogMsg(const debug_report_data *debug_data, VkFlags msg_flags, const LogObjectList &objects, const Location *loc,
-                       std::string_view vuid_text, const char *format, va_list argptr);
+  private:
+    bool UpdateLogMsgCounts(int32_t vuid_hash) const;
```
Comment (spencer-lunarg) [inline layers/error_message/logging.h:231]: these were `static` but still required passing in a pointer, so just made them private functions

File: layers/error_message/logging.h [hunk 3]
```diff
@@ -334,109 +277,3 @@ VKAPI_ATTR VkBool32 VKAPI_CALL MessengerWin32DebugOutputMsg(VkDebugUtilsMessageS
                                                             VkDebugUtilsMessageTypeFlagsEXT message_type,
                                                             const VkDebugUtilsMessengerCallbackDataEXT *callback_data,
                                                             void *user_data);
-
-template <typename Map>
-static LoggingLabelState *GetLoggingLabelState(Map *map, typename Map::key_type key, bool insert) {
```
Comment (spencer-lunarg) [inline layers/error_message/logging.h:339]: things like this I just moved, it is static, but everyone using it is in `logging.cpp` so no need to have it out in the header file

File: layers/error_message/logging.h [hunk 4]
```diff
@@ -260,67 +214,56 @@ typedef struct _debug_report_data {
...
+    bool LogMsg(VkFlags msg_flags, const LogObjectList &objects, const Location *loc, std::string_view vuid_text,
+                const char *format, va_list argptr);
 
-template debug_report_data *GetLayerDataPtr<debug_report_data>(void *data_key,
-                                                               std::unordered_map<void *, debug_report_data *> &data_map);
+    void BeginQueueDebugUtilsLabel(VkQueue queue, const VkDebugUtilsLabelEXT *label_info);
+    void EndQueueDebugUtilsLabel(VkQueue queue);
+    void InsertQueueDebugUtilsLabel(VkQueue queue, const VkDebugUtilsLabelEXT *label_info);
 
-static inline void DebugReportFlagsToAnnotFlags(VkDebugReportFlagsEXT dr_flags, VkDebugUtilsMessageSeverityFlagsEXT *da_severity,
```
Comment (spencer-lunarg) [inline layers/error_message/logging.h:268]: this was just moved into `logging.cpp`

File: layers/core_checks/cc_synchronization.cpp [hunk 5]
```diff
@@ -1949,9 +1949,8 @@ void CoreChecks::RecordBarriers(Func func_name, vvl::CommandBuffer &cb_state, co
 }
 
 template <typename TransferBarrier, typename Scoreboard>
-bool CoreChecks::ValidateAndUpdateQFOScoreboard(const debug_report_data *report_data, const vvl::CommandBuffer &cb_state,
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_synchronization.cpp:1952]: `report_data` was unused here

File: layers/error_message/logging.h [hunk 6]
```diff
@@ -178,76 +178,30 @@ class TypedHandleWrapper {
     VulkanTypedHandle handle_;
 };
 
-typedef struct _debug_report_data {
+struct Location;
+
+class DebugReportData {
```
Comment (artem-lunarg) [inline layers/error_message/logging.h:183]: nit: real nit about Data suffix, not sure if it's helpful.

**Review by artem-lunarg — APPROVED** [review]
LGTM

File: layers/error_message/logging.h (hunk 6 above)
Comment (spencer-lunarg) [reply layers/error_message/logging.h:183]: it isn't.... let me fix it now as it will be too nit to ever touch again lol

**Pull request merged**